### ISSUE TITLE
[Bug] Mouse is jumping

### ISSUE THREAD

**Issue by domske** [debauchee/barrier#300]
#### Bug Report

The mouse is jumping if moved from rest. No problem when I move it continuously.
But if you wait between movements (just a few milliseconds / < second), the mouse jumps.
It's like it sleeps immediately, as soon as she stops moving.

See screen record:
![screenrecord](https://user-images.githubusercontent.com/28209992/57329434-0a716900-7114-11e9-88ce-5b99ec1a35bf.gif)

This is very annoying. I expected a smooth movement. This is not possible if the mouse awakes delayed and jumps to position.
Maybe this is a wanted behavior to hibernate the transmission. If so, the timeout time should be increased to a minute or a few seconds.

#### Version
Server: Windows 10 Home (2.2.0)
Client: Mac OS X High Sierra (2.10)

Any questions? I am glad to help.

**Comment by noisyshape** [comment]
Would I be correct in assuming that you're connecting over wifi? And that one or both systems have a power saving feature enabled? A mitigation for this might be sending dummy packets for a while after mouse movement.

**Comment by domske** [comment]
Yes, both devices are connected via WiFi in same network on a router. Do you mean, using LAN will not have this problem?...

**Comment by mrwensveen** [comment]
I also have this issue when I'm connected over WiFi. It's not consistent, though. Sometimes everything is silky smooth and sometimes not. I think it might depend on how much the network is utilized (it definitely seems worse when the kids are home and streaming video etc.)

**Comment by AdrianKoshka** [comment]
New release, unlikely it fixes this issue, but I thought I'd bring it up: https://github.com/debauchee/barrier/releases/tag/v2.3.0

**Comment by maliesa96** [comment]
I had this issue. Decreasing the polling frequency on my mouse fixed it.

**Comment by tinybot-ca** [comment]
I recall this being an issue with Synergy way back in the day -- and connecting my client machine (Macbook Pro laptop) via Ethernet got around it. Now, many years later, this still seems to be an issue on Wi-Fi? I suppose I can dig up an ethernet dongle -- but would love for a solution that would work for Wi-Fi.

**Comment by pinei** [comment]
It seems to perform better over 5 Ghz Wifi than 2 Ghz

**Comment by zhongke-chen** [comment]
I also had the issue. I solved this by disabling both MSDU & MPDU aggregation on my wifi router.

**Comment by mrpew** [comment]
Bug still relevant, I get frequent mouse stuttering/jumping between two Ubuntu Systems connected via **Ethernet** with a sub 1ms ping. 

Sometimes happens occasionally (e.g. every 10-30 seconds) sometimes constantly (Unusable until barrier process is killed or direct input on the client is given).

While its happening, using the keyboard does not work either.

**Comment by fiftyfathoms** [comment]
This bug affects me too. Kubuntu host 18.04.5, Windows 10 guest. It's strange that everything used work just fine for months and all of a sudden the mouse pointer started to show extreme stutters today.

**Comment by Skrimpton** [comment]
I have the same issue on one manjaro nibia 20.2 client.

It occurs with both a wired mouse and a wireless mouse. 
It also stops even with continous movement

It is not an issue (at all) on a clean-installed Manjaro minimal nibia 20.2 laptop, where both share a windows 10 host.

The client with the bug is an upgraded system, that has been in use for about  6/7 months, with several "extra" apps installed.
The bug occurs with both a wireless logitech g602 mouse and a wired hp M-BT96a. Keyboard (logitech k350 wireless) is seemingly not affected.

**Comment by clankill3r** [comment]
Same issue with 10.14.6. Ethernet cable made it better, but that's not really an option for me.

**Comment by fiftyfathoms** [comment]
An update to my previous comment:
After also checking Synergy and experiencing the same issue, I changed the the wifi band of my router from 2.4 to 5 GHz and the issue seems to be gone. I had switched off the 5 GHz channel long before, mainly due to the rather long distance between my computer and the router. Apparently even slight change in the position of the router helps a great deal with the transmission performance of the signal.

**Comment by shridhar-hegde** [comment]
I still have this issue using Windows 10 -> Linux Lite connected over wifi. 
Many of you have suggested to switch to 5GHz band but my router does not have that band!
Is there any other fixes that any of you have found?

When I am using Windows 10 -> Windows 10, Mouse Without Borders works flawlessly !

**Comment by hungerregnuh** [comment]
For my setup:
  - Client is a MBP (Catalina and Big Sur)
  - Linux Mint as my server and had no issues
  - I'm now running it as a server on a MBP (Big Sur) and this is when i noticed the mouse jumping issue.

What ultimately fixed it for me was getting my multi-display issue fixed.  Anyone running a MBP with multi-display knows how finicky OSX has been lately with them.  I have a 3 display setup and getting the third display to be stable was key.  Once i got the third display to stop reconnecting the mouse jumping issue went away.  Even when i don't have the third display connected the mouse jumps.  I suspect barrier, OSX or both are "remembering" the third display even though it's not connected and is causing the mouse to jump around as something is constantly changing the actual width of my desktops.

not sure if this helps, hope it does.

**Comment by hungerregnuh** [comment]
Not a fix, but it works - i also found out that if you enable the double tap edge to switch, the mouse doesn't jump around anymore, not as convenient sure - but works.

**Comment by LeonardSSH** [comment]
Has anyone solved this problem? (🟦Windows > 🐧Ubuntu) 😊

**Comment by jwallet** [comment]
Maybe the core team could give more detail

Both of my computers are on LAN (windows and mac) and I got the same behavior, only when Rider or Vscode was running foreground.

But it seems to be better after doing the following changes in the server configuration 
- Click _Configure server..._
- _Advanced server settings_ tab
- I checked : _Check clients every_ and _use  relative mouse moves_

It seems a lot better now

**Comment by bbook1997** [comment]
Running Windows 10 Pro 10.0.19041 Build on two PC's.   Connected via wireless for months now.
Running Barrier 2.3.3-release-f2bf7cb7 without any issues during all this time except for one time with mouse became sluggish on client.   Restarting the server fixed it.   Been running fine for weeks since then.
Suddenly, yesterday, the mouse has become very choppy on the client.   I've rebooted both server and client twice since then, and it is still the same.  I tried @hungerregnuh 's workaround, but that didn't help.  I've tried changing the port used, but that hasn't helped.   Logging is not turned on.   Over 100GB available on all disk drives.  
Would love to get some ideas on how to fix it.

I believe it is related to the fact that I have been doing a lot of testing using applications that talk to each other over the same network, and yesterday I had them sending tons of messages back and forth.   But that test is not running now, and after rebooting, I had hoped it would clear out whatever is causing the delays, but nothing has helped so far.

**Comment by harveyr** [comment]
I just had this problem for the first time. Host is macOS, client is Ubuntu. The problem seemed to coincide with a large-ish Google Meet on the client machine, which I assume caused a lot of network activity. More troubling than the mouse choppiness was the keyboard activity. It acted like keys were sticking on the client machine. So I'd try to type "Hello" and end up with "Hhhhhhhhhhheloooooooooo."

The problem persisted after the meeting ended. I tried restarting both client and server to no avail. I used the Reload button and hadn't tried manually restarting the processes yet. But just as I arrived here to write this, the problem went away. 🤷

**Comment by noveltomatillo8** [comment]
I'm having this same issue for the first time on my client. Host and client are both W10, WiFi host, ethernet client. The (wireless) mouse cursor lags and skips around for 1-5 seconds every 5-10 seconds no matter what speed I move the cursor but keyboard is totally fine. Switching mice didn't make a difference. Did a clean reinstall of Barrier on both machines to no avail. I only have one band on my router, so changing that isn't an option.

Worked perfectly for a month or so, but now randomly seeing this issue with no changes to either machine that I can think of. Any idea if this is just an underlying bug, or did any fix make a lasting difference?

**Comment by bbook1997** [comment]
Somewhat off topic, but I switched over to "Mouse Without Boarders" from Microsoft.   And so far, I've had very little problem with it.  Only once when I had heavy internet traffic did the mouse start lagging, but a quick disconnect / reconnect fixed it immediately.  Also, I had a lot of problems with Barrier connecting my two computers together.   Sometimes having to fiddle with things for several minutes.   I've had zero issues with MWB connecting the two PC's together.   Also, it's a 2-way connection, so I can use either mouse or keyboard.   All in all, much better experience on Windows 10

**Comment by intositeme** [comment]
> Somewhat off topic, but I switched over to "Mouse Without Boarders" from Microsoft. And so far, I've had very little problem with it. Only once when I had heavy internet traffic did the mouse start lagging, but a quick disconnect / reconnect fixed it immediately. Also, I had a lot of problems with Barrier connecting my two computers together. Sometimes having to fiddle with things for several minutes. I've had zero issues with MWB connecting the two PC's together. Also, it's a 2-way connection, so I can use either mouse or keyboard. All in all, much better experience on Windows 10

Thanks for the share! Was connecting 2 Windows 10 systems, one on LAN and another on Wifi, and had the jumpy/laggy mouse on the client system. Tried "Mouse Without Boarders" and it didn't have the same kind of lag.

**Comment by jwallet** [comment]
I'm still following this thread, but I found out it was better for me to be connected by LAN instead of wiressly (WAN). I'm sure I will always have issues on wifi because my laptop only connects to the 2.4gHz network, and I also tried to have everything on the same desk (pc (wired), laptop (wifi), router) and I still got issues.

So I just want to leave a friendly reminder, that 2.4gHz frequency band network can have their beacon signal interfered by many stuff in your house that use this same frequency, like appliances, like microwave or the router next door or even your cellphone,  unlike 5gHz that have less interference. However 5gHz are easily stop by walls, so they reach less far than 2.4gHz band network.

**Comment by clankill3r** [comment]
I don't think people realize how fast computers are. You don't need 5gHz in order to send the mouse position 60 times a second.

**Comment by jwallet** [comment]
I did not talk about the speed, but interference, I'm talking about shitty home router and having to many stuff on the 2.4gHz network, that may lower the quality of your wireless signal randomly. I want this software to be accurate at all time, not dropping for a second every 15 minutes or so, or taking 5 seconds to track the mouse when the laptop is waking up and reconnecting to the wifi

**Comment by idenc** [comment]
Even on a 5gHz network, it stutters for me. Only solution was to use LAN

**Comment by aboode95** [comment]
To make barrier work over wifi, disable wifi power management:

Run:

    sudo nano /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf

This conf file contains: 

    [connection]
    wifi.powersave = 3

change the value to '2'. And reboot.

These instruction are from [here](https://unix.stackexchange.com/questions/269661/how-to-turn-off-wireless-power-management-permanently).

Big thanks to @noisyshape for mentioning it is a wifi power saving feature issue!

(I am using a laptop with ralink 3090 card. Pretty troublesome card I must say)

**Comment by sgon00** [comment]
I am running barrier in three devices: Linux (Debian 10), Windows 10, MacOS (Big Sur).
Linux as the server.
Windows 10 and MacOS as the clients.
I don't have any problems btw Linux and Windows 10.
The lagging issue only happens in MacOS Client. 
To me, the problem is only in MacOS.

**Comment by kvnxiao** [comment]
If you don't need AirDrop, disable the `awdl0` network interface which is for Apple Wireless Direct Link:
```
sudo ifconfig awdl0 down
```

This will help alleviate the choppiness in the mouse movement _a lot_

**Comment by nocode99** [comment]
I'm using the server on Big Sur and Ubuntu 20.04 as the client (I also reversed the set up as well) and none of the recommendations helped. I unfortunately cannot test using wired connections but found disabling `Wi-Fi Networking` in Location services to get rid of the issue.  There's a little bit of jitter, but I don't expect the client to be 100% smooth.

Go to *System Preferences -> Security & Privacy -> Location Services* Unlock on the bottom left corner and then select `Details` in System Services. Uncheck `Wi-Fi Networking`

![disable](https://i.imgur.com/9tnFqWL.png)

**Comment by xQsme** [comment]
Hi, I believe I'm suffering from this same issue.

I'm running the server on Linux and accessing on a MacOS client.
I've tried every suggestion except for plugging in an ethernet cable (don't have a dongle).

In my case, this issue was very severe, my laptop would freeze constantly, about 2s every 10s, during zoom calls.
After testing my internet stability (https://test.vsee.com/network/index.html), I bought a pretty good Wi-Fi powerline, and my connection became very stable, with low ping, high bandwidth, and no packet loss.

Currently, when listening to music, watching videos, or during zoom calls, I might occasionally get a short freeze (less than 1 sec), where my audio glitches completely.

If i click on the Wi-Fi indicator on MacOS's menu bar, the mouse movement loses it's smoothness for a couple seconds, I've also re-ran the network stability tests while doing this action and I end up with 8-40% packet loss when repeatedly doing this action, though if I disconnect Barrier and use my laptop's touchpad, the connection is flawless.

I'd honestly appreciate it a lot if this issue was resolved, and I'm available to contribute with anything I'm able to.
Thanks.

**Comment by webdog** [comment]
I'm experiencing this issue on a hard-wired setup. Each machine is on the same switch so latency is sub 1ms. No error messages reported in the log (logs turned up to highest debug level), but the mouse appears to freeze/jump every 100-500px without fail. Keyboard also suffers from latency. Issues survive a restart, and I've also swapped client/server with no resolution. On each host machine, the keyboard and mouse work fine.

MacOS 11.6 on both machines.

**Comment by itsCaveMan** [comment]
I solved my lag by changing client sync to 100ms

`Configure server --> Advanced server settings --> Check clients every [100]ms`

![Screenshot 2021-10-17 at 16 41 30](https://user-images.githubusercontent.com/51651537/137632404-76dc35e7-de8f-412f-9a0a-64dc19fd8562.png)


I have no lag on my 5g wifi AP, and the odd second here and there on my none-5g wifi AP

Host: MacOS Catalina
Client: Windows 10

**Comment by yk67** [comment]
Could be entirely anecdotal, but connecting to a shared drive on the host machine via smb from my mac seemed to stabilize the connection—even on wifi.

**Comment by RainOfPain125** [comment]
Waiting for devs to fix this, 3 years later.

I'm having this problem. I skimmed all the discussion, unlike others I am using a straight up ethernet connection, so wifi signal or what not is not affecting this at all.

I tried what some people here suggested - changing "Use relative mouse moves" on, and lowering the "Check clients every" from 5000ms to 1000ms, even 100ms. None of these fixed it.

Barrier was working just fine when I first started using it, I changed none of my settings. This leads me to believe there is nothing wrong on my end, and that Barrier simply starts to fuck up over time? I mean, I'm new to Linux and I've been installing stuff.

I installed Samba and setup a computer-to-computer file share over the same ethernet cord. However, disabling Samba (sudo service smbd stop, sudo service nmbd stop) did not fix the problem either (as I suspected maybe Samba was messing up the connection in some way). I think the problem may have been eased a little, but its probably just the placebo effect of turning off what I thought was harming Barrier's performance. Nevertheless, the problem still persists.

I used "iftop" to monitor all the ethernet connections between the two computers. Even when Barrier was the only connection, the input leaps still persisted.

**Comment by clankill3r** [comment]
I stopped using Barrier and use ShareMouse now.

**Comment by nuttakornbe** [comment]
Windows<->Mac
Turn off universal control and disable handoff feature works for me. 
![image](https://user-images.githubusercontent.com/16710825/149873347-31a86185-7cdf-427f-992f-940ab86f3ef9.png)

**Comment by RainOfPain125** [comment]
> I stopped using Barrier and use ShareMouse now.

Sadly this is not an option for me, because ShareMouse does not support Linux.

**Comment by mooreaa** [comment]
Am having this issue as well.

Host: Mac OS Big Sur
Client: Win 11

Both machines are connected via ethernet. Mouse will periodically jump in normal use. 

I have tried ShareMouse and it doesn't have the same problem. 

I'm wondering if somehow its a process priority issue where barrier is being starved under heavy load conditions. 

---- Using ShareMouse is NOT a good alternative as its 150 for the license ---- Hoping to find out why we're having this jumping issue with Barrier.

**Comment by mehdico** [comment]
I fix this by change port from 24800 to 24801 :)
Mac OS <> Manjaro with Gnome

**Comment by Agustin-Q** [comment]
> It seems to perform better over 5 Ghz Wifi than 2 Ghz


For me this solved the problem 100%

**Comment by j2L4e** [comment]
For most people the issue should be their wireless adapter's powersave mode. When you continouusly ping your laptop, you'll likely notice rising latency when there's no network activity.

I don't think solving this on barrier's side (without effectively breaking wifi powersaving) is straight-forward.

My laptop usually runs on AC power when I use barrier, so I'm disabling powersave mode when not on battery.

To automate this on arch linux I'm using these udev rules. Attribute and device names can be different on other systems.
```
SUBSYSTEM=="power_supply", ATTR{online}=="0", RUN+="/usr/bin/iw dev wlp1s0 set power_save on"
SUBSYSTEM=="power_supply", ATTR{online}=="1", RUN+="/usr/bin/iw dev wlp1s0 set power_save off"
```

**Comment by sgon00** [comment]
@j2L4e I totally disagree with you. You can only determine your case. How can you determine the cause for most people? My case is not related to wireless adapter's powersave mode at all.  It only happens in MacOS. You can see that the author is having the problem in MacOS too. I don't know what happened to your arch linux, but the author is not using arch linux for sure.

**Comment by j2L4e** [comment]
> You can only determine your case. How can you determine the cause for most people? 

Enabling wifi powersaving reproducibly causes the exact behavior described by the author. It is also a useful feature that's enabled on close to all platforms by default. So if you're experiencing the described problem, this is a likely cause.

> It only happens in MacOS.

That's obviously untrue because I observed the same thing on linux.

> You can see that the author is having the problem in MacOS too. I don't know what happened to your arch linux, but the author is not using arch linux for sure.

The suspected cause is platform-independant. Letting people know how I worked around the problem might help them fix it on their end.

> I totally disagree with you.

That's okay. 
Why do you think what I wrote doesn't fit your problem?

**Comment by emircanerkul** [comment]
> I'm still following this thread, but I found out it was better for me to be connected by LAN instead of wirelessly (WAN). I'm sure I will always have issues on wifi because my laptop only connects to the 2.4gHz network, and I also tried to have everything on the same desk (pc (wired), laptop (wifi), router) and I still got issues.
> 
> 2.4gHz frequency band network can have their beacon signal interfered by many stuff in your house that use this same frequency, like appliances, like microwave or the router next door or even your cellphone, unlike 5gHz that have less interference. However 5gHz are easily stop by walls, so they reach less far than 2.4gHz band network.
> 
> It might not be what cause your problem, though. But, if someone wants to try out the app using 5.gHz band network, that would be nice.

I'm here because I had the same problem. After I changed to 5ghz the problem disappear

**Comment by caseyrapnicki** [comment]
I had Debug2 logging enabled (only to the error log, not to a file) on my client laptop (MacOS). Once I turned it back to Info, the mouse lag went away.

**Comment by spacegaucho** [comment]
> For most people the issue should be their wireless adapter's powersave mode. When you continouusly ping your laptop, you'll likely notice rising latency when there's no network activity.
> 
> I don't think solving this on barrier's side (without effectively breaking wifi powersaving) is straight-forward.
> 
> My laptop usually runs on AC power when I use barrier, so I'm disabling powersave mode when not on battery.
> 
> To automate this on arch linux I'm using these udev rules. Attribute and device names can be different on other systems.
> 
> ```
> SUBSYSTEM=="power_supply", ATTR{online}=="0", RUN+="/usr/bin/iw dev wlp1s0 set power_save on"
> SUBSYSTEM=="power_supply", ATTR{online}=="1", RUN+="/usr/bin/iw dev wlp1s0 set power_save off"
> ```

yes this seems to be the problem. tried it out in linux client (lenovo notebook): `echo 'on' | sudo tee /sys/bus/pci/devices/0000:3d:00.0/power/control`

no more stutters, and as a bonus: latency decreased and mouse wheel doesn't stop working anymore.

if you are using linux, try using `sudo powertop` (needs to be installed in most distros) and figure out which wireless adapter you have. then disable powersaving.

**Comment by minimask** [comment]
> If you don't need AirDrop, disable the `awdl0` network interface which is for Apple Wireless Direct Link:
> 
> ```
> sudo ifconfig awdl0 down
> ```
> 
> This will help alleviate the choppiness in the mouse movement _a lot_

Thanks! This fixed it for me. Worth a try if you're sharing between two Macs.

**Comment by mishmishb** [comment]
@j2L4e @spacegaucho do either of you maybe have an idea of how to do this for MacOS? I would like to try and see if it helps as well :) however I haven't been able to find a useful result from Google

**Comment by mishmishb** [comment]
> If you don't need AirDrop, disable the `awdl0` network interface which is for Apple Wireless Direct Link:
> 
> ```
> sudo ifconfig awdl0 down
> ```
> 
> This will help alleviate the choppiness in the mouse movement _a lot_

This helped me a lot too :) so I guess this was the main reason for my issues

**Comment by aryansaurav** [comment]
There could be multiple reasons for the lag and some of the hacks mentioned above might help.
For me, I unchecked "use relative mouse cursor" in the barrier Configure server. Helped speed up quite a bit.

**Comment by MisutaaAsriel** [comment]
@itsCaveMan's suggestion to decrease the client check period worked to reduce the lag but it's *not* gone.

The PC client is hardwired but the MacBook Air is on 5GHz Wi-Fi. The lag is also really intermittent.

I'm not entirely sure what may be causing it in my cases. — Changing the ethernet adapter's power saving options helped a little too, though.

**Comment by glantucan** [comment]
I can confirm that in my case, both server and client being linux (pop_os! 22.04) and both conneted through Wi-Fi, disabling power saving for the Wi-Fi solved the issue completely. 
Steps to disable Wi-Fi power saving:
* Edit `/etc/NetworkManager/conf.d/default-wifi-powersave-on.conf` as root.
  ```
  sudo nano /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf
  ```
* Change `wifi.powersave = 3` to `wifi.powersave = 2`
* Run `systemctl restart NetworkManager`

**Comment by draeder** [comment]
I added port 28400 to my WiFi router's QoS and gave it highest priority. I then also configured the Barrier server's advanced settings to check clients every 5000 ms. The two together seems to alleviate my issues. However, I've had false positives on a fix for this in the past.

What I found by inspecting the network traffic is that the packets are being retransmitted when the lag occurs. They are TCP packets, which in my opinion is the wrong traffic type for this application. To me, it should be UDP which is better suited for real-time streaming data.

**Comment by annetimdegraaf** [comment]
I'm running the latest barrier on a Mac Mini and a Windows 11 laptop. The Mac Mini is the server. Both are connected with ethernet to the same switch. Mouse warps like crazy on the Windows 11 Client. Tried everything from this list. Disable Airdrop, disable Relative Mouse, and enabled. Changed the "Check client every 1000ms". Nothing helps. The Windows 11 client just won't work like it should. Wished it was a WiFi problem but in my case, this is not the fix.

**Comment by aryansaurav** [comment]
> I'm running the latest barrier on a Mac Mini and a Windows 11 laptop. The Mac Mini is the server. Both are connected with ethernet to the same switch. Mouse warps like crazy on the Windows 11 Client. Tried everything from this list. Disable Airdrop, disable Relative Mouse, and enabled. Changed the "Check client every 1000ms". Nothing helps. The Windows 11 client just won't work like it should. Wished it was a WiFi problem but in my case, this is not the fix.

I am also using it on Mac Mini and Windows PC. The way it worked for me was to run barrier in server mode on the Mac and client mode on Windows. You may have to specify the server's IP on the Windows PC but sometimes it may get detected automatically. 

I didn't need any other trick. But at the same time, the other way (running server on Windows and client on Mac) never works at all. Hope this helps!

**Comment by annetimdegraaf** [comment]
> > I'm running the latest barrier on a Mac Mini and a Windows 11 laptop. The Mac Mini is the server. Both are connected with ethernet to the same switch. Mouse warps like crazy on the Windows 11 Client. Tried everything from this list. Disable Airdrop, disable Relative Mouse, and enabled. Changed the "Check client every 1000ms". Nothing helps. The Windows 11 client just won't work like it should. Wished it was a WiFi problem but in my case, this is not the fix.
> 
> I am also using it on Mac Mini and Windows PC. The way it worked for me was to run barrier in server mode on the Mac and client mode on Windows. You may have to specify the server's IP on the Windows PC but sometimes it may get detected automatically.
> 
> I didn't need any other trick. But at the same time, the other way (running server on Windows and client on Mac) never works at all. Hope this helps!

Thanks for helping me! I use the Mac Mini as server. Still warps on my client computer. So same setup but the with problems on my end. The connection between the two works, that is not the problem. It's just not stable on the client side.

**Comment by minimask** [comment]
Thought disabling Airdrop fixed it for me, but the problem came back.  I read on the Synergy site that it's a known Apple bug that Apple isn't prioritizing to fix. More info [here](https://symless.com/synergy-help/what-to-do-when-the-mouse-cursor-lags).

**Comment by draeder** [comment]
@minimask I have had this problem with all Synergy-based vKVMs between every OS I've ever tried to use them on. It has been this way for over a decade. I do not believe this is a problem with a particular OS. TCP should really not be used for realtime data. There is a reason multiplayer FPS games use UDP.

**Comment by MisutaaAsriel** [comment]
> @minimask I have had this problem with all Synergy-based vKVMs between every OS I've ever tried to use them on. It has been this way for over a decade. I do not believe this is a problem with a particular OS. TCP should really not be used for realtime data. There is a reason multiplayer FPS games use UDP.

Not only that, but Synergy says it's an issue primarily effecting Mac *clients*, but for myself, and I'm sure many others, we're running the server from our Mac.

After all, which is more likely to be ran partially "headless" (handless?) — A desktop PC with no mouse or keyboard, nestled in the corner of the living room plugged in to a TV? Or your expensive fruit-company laptop that you can throw in a bag and take with you everywhere?

it's based on an assumption that users *prefer* PC, which says no one is really quite testing their software the other direction.

**Comment by minimask** [comment]
@MisutaaAsriel yes, I'm also running both the client and the server on two M1 Macs.

**Comment by draeder** [comment]
@minimask  @mishmishb 
I've run it between mac <-> windows, windows <-> linux, mac <-> linux, with the server on both sides of all three and all three had the same issues. Since at least 2009.

**Comment by minimask** [comment]
@draeder does it happen only over Wifi?  I'm not close to my router so could not test it via ethernet.

**Comment by draeder** [comment]
@minimask  Unfortunately, I'm in the same boat. I did try a local switch some years ago and it didn't make a difference if I recall correctly. At that time it was windows <-> windows.

**Comment by vjpr** [comment]
Anyone notice this is a new issue?

I'm running `macOS (server - macbook) -> Windows 10 (client - 2nd macbook bootcamp)`.

I haven't used Barrier for a couple of months, but I remember it being really smooth and flawless.

I was using Mac to Mac recently with `Universal Control` and so had to enable AirDrop, etc.

Maybe Universal Control or some recent mac update it causing it.

**Comment by draeder** [comment]
I've been able to confirm a reliable workaround for macOS (server) -> Windows 10 (client). It isn't a permanent workaround, but when the issue crops up it does seem to be fairly reliable.

Mac:
1. Stop Barrier
2. From System Preferences, choose Network > WiFi
3. Click Turn WiFi Off
4. Click Advanced
5. Click Hardware
6. Change the Configure dropdown from Automatically to Custom
7. Click OK
8. Click Apply
9. Turn WiFi On
10. Repeat these steps but change the Configure dropdown from Custom back to Automatically
11. Start Barrier

This seems to work for me every time the issue crops up. The steps are tedious but not as tedious as the mouse jumping all over and keystrokes being delayed. 

Please report your mileage with this.

**Comment by lk251** [comment]
> I've been able to confirm a reliable workaround for macOS (server) -> Windows 10 (client). It isn't a permanent workaround, but when the issue crops up it does seem to be fairly reliable.
> 
> Mac:
> 
> 1. Stop Barrier
> 2. From System Preferences, choose Network > WiFi
> 3. Click Turn WiFi Off
> 4. Click Advanced
> 5. Click Hardware
> 6. Change the Configure dropdown from Automatically to Custom
> 7. Click OK
> 8. Click Apply
> 9. Turn WiFi On
> 10. Repeat these steps but change the Configure dropdown from Custom back to Automatically
> 11. Start Barrier
> 
> This seems to work for me every time the issue crops up. The steps are tedious but not as tedious as the mouse jumping all over and keystrokes being delayed.
> 
> Please report your mileage with this.

this solved the issue but after my mac, the server, went to sleep, the client, a win11 pc, started having the issue again.

how to solve this permamently? it didn't happen until today, i had been using barrier with no issues for a few days before this.

**Comment by draeder** [comment]
> 
> this solved the issue but after my mac, the server, went to sleep, the client, a win11 pc, started having the issue again.
> 
> how to solve this permamently? it didn't happen until today, i had been using barrier with no issues for a few days before this.

Unfortunately, there is no permanent fix. It has been an issue since at least 2009 and appears that it will never be solved. It's so bad, I have been looking into writing my own mouse and keyboard sharing software.

**Comment by nuttakornbe** [comment]
I want to share an approach that it's working for me after I used Barrier for years and after tried all possible solutions on this thread and others.

Try "turning off all logging on all computers running Barrier, the severs and all the clients." 

IO (log read/write) operation is costly in general and sometimes impacting the mouse movement especially triggered in the same app like Barrier (even running in different threads).

**Comment by Congles** [comment]
I just installed Barrier and ran into this issue with a Mac -> Windows hardware setup.

### **enabling _"Switch after waiting"_ in the Advanced Server settings area fixed this issue entirely for me.** 

The minimum setting seems to be 25ms and is an almost entirely unnoticeable delay. I wonder if the Dev's could look at this as a potential fix - perhaps by having a check box _"Fix mouse jumping (enable if you have trouble with mouse teleports)"_ and this might set the value internally to 5ms.

**Comment by lk251** [comment]
> > this solved the issue but after my mac, the server, went to sleep, the client, a win11 pc, started having the issue again.
> > how to solve this permamently? it didn't happen until today, i had been using barrier with no issues for a few days before this.
> 
> Unfortunately, there is no permanent fix. It has been an issue since at least 2009 and appears that it will never be solved. It's so bad, I have been looking into writing my own mouse and keyboard sharing software.

thanks for the heads up. if you do write it, please do share a link here, i'd love to try it!

**Comment by PrafulDhungana** [comment]
I don't know if i could call this an actual solution but,  when I plug in my client side laptop there are no stutters, and when I pull the plug the mouse stutters are back.
Windows 7 Desktop Host and Windows 10 Laptop client.

**Comment by draeder** [comment]
> The minimum setting seems to be 25ms and is an almost entirely unnoticeable delay. I wonder if the Dev's could look at this as a potential fix - perhaps by having a check box _"Fix mouse jumping (enable if you have trouble with mouse teleports)"_ and this might set the value internally to 5ms.

This appears to have greatly reduced the delay I was experiencing (notably I was able to lower this all the way to 10ms). It still happens, but not to the degree where it happens every half second (or less) with the delay sometimes happening for 1-2 seconds over minutes to hours at a time.

**Comment by mikbuch** [comment]
> I just installed Barrier and ran into this issue with a Mac -> Windows hardware setup.
> 
> ### **enabling _"Switch after waiting"_ in the Advanced Server settings area fixed this issue entirely for me.**
> The minimum setting seems to be 25ms and is an almost entirely unnoticeable delay. I wonder if the Dev's could look at this as a potential fix - perhaps by having a check box _"Fix mouse jumping (enable if you have trouble with mouse teleports)"_ and this might set the value internally to 5ms.

Amazing, this solution helped me to with Mac as a server and (an older) Mac as a client. Although this did not resolved the problem entirely, the lagging is now acceptable.

Btw., here is how to access advanced Barrier settings: https://github.com/debauchee/barrier/issues/232#issuecomment-455831297

**Comment by rayanfer32** [comment]
> I can confirm that in my case, both server and client being linux (pop_os! 22.04) and both conneted through Wi-Fi, disabling power saving for the Wi-Fi solved the issue completely. Steps to disable Wi-Fi power saving:
> 
> * Edit `/etc/NetworkManager/conf.d/default-wifi-powersave-on.conf` as root.
>   ```
>   sudo nano /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf
>   ```
> * Change `wifi.powersave = 3` to `wifi.powersave = 2`
> * Run `systemctl restart NetworkManager`

Thank you , I was using windows 8 as host and ubuntu 22 laptop as client , and this solution solved the multiple keystrokes issue and mouse lag

**Comment by akayuki39** [comment]
I am using a mac as sever and a win as client in a 5G WIFI environment. It used to work well but I found the mouse become jumpy and laggy today. 
Eventually find the reason is that my old mac's WIFI is not working well. Disable and enable the WIFI switch on my mac solve this problem. Hope this helps.

**Comment by Henriquex25** [comment]
> Posso confirmar que no meu caso, tanto o servidor quanto o cliente sendo Linux (pop_os! 22.04) e ambos conectados via Wi-Fi, desabilitar a economia de energia para o Wi-Fi resolveu o problema completamente. Etapas para desativar a economia de energia do Wi-Fi:
> 
> * Edite `/etc/NetworkManager/conf.d/default-wifi-powersave-on.conf`como root.
>   ```
>   sudo nano /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf
>   ```
> * Mudar `wifi.powersave = 3`para`wifi.powersave = 2`
> * Correr`systemctl restart NetworkManager`

This solved my problem!!! Thankss

W11 > Pop_OS 22.04

**Comment by gregwidener** [comment]
I'm using Windows 11 as the server and a Steam Deck OLED as the client. I had the jumpy mouse issue and went through the entire thread and tried every trick as far as settings go. None of that worked. What worked for me was disabling wifi power savings on the Steam Deck, which I found in another forum. You have to go to Settings -> System -> Enable Developer Options. Once that is enabled, Developer Options are at the bottom of Settings. Go in there and disable Wifi Power Savings, which requires a restart. 100% fixed it for me! Looks like the issue is power savings on the wifi adapter across multiple versions of Linux/Unix/Mac OS X.

**Comment by 4nacs4** [comment]
> I just installed Barrier and ran into this issue with a Mac -> Windows hardware setup.
> 
> ### **enabling _"Switch after waiting"_ in the Advanced Server settings area fixed this issue entirely for me.**
> The minimum setting seems to be 25ms and is an almost entirely unnoticeable delay. I wonder if the Dev's could look at this as a potential fix - perhaps by having a check box _"Fix mouse jumping (enable if you have trouble with mouse teleports)"_ and this might set the value internally to 5ms.

You rock, this solution helped me a lot with my macos issues

**Comment by ekoleszar** [comment]
> An update to my previous comment: After also checking Synergy and experiencing the same issue, I changed the the wifi band of my router from 2.4 to 5 GHz and the issue seems to be gone. I had switched off the 5 GHz channel long before, mainly due to the rather long distance between my computer and the router. Apparently even slight change in the position of the router helps a great deal with the transmission performance of the signal.

That worked for me thank you very much!

**Comment by kr1surb4n** [comment]
Start from the network settings. 
Check your router first. You might have some default settings which mess up the connection. 
i.e. 
- my new router had the long preamble on by default -> the packets contain additional error correction data and are bigger, short preamble fixed the issue. 
- the 5GHz can solve the issue too, as it can send more data per unit of time
- you might have also some sort of packet filtering 
After network check your system. You might also change which machine is the server and the client. 
If nothing works revert to cable.

**Comment by xortuna** [comment]
This issue purely stems from WIFI's power saving modes.

If you are on Mac no option exists to change the WIFI power saving feature,

Instead you have to utilise the connection constantly to stop it going into power save mode, apple is **VERY** aggressive with their power saving, if a packet hasn't been sent in 200ms it'll power save and introduce a 300ms delay before the adapter will even begin acknowledge packets again.

To solve this simply ping your server every 200ms and it will reset the timeout and prevent the wifi adapter going into powersave, and eliminate the lag completely.
`ping [YourServerIp] -i 0.20
`

**Comment by xtremextr** [comment]
Finally fixed this problem on my end, ironically by a completely unrelated reason - buying a new mouse.

I'm using Barrier with Windows 11 as server x Linux Mint as client
Mouse stuttering was unbearable, I had to move it very slowly and steadily to have no stuttering.

Both are laptops using the same Wi-Fi network, tried everything listed here before, deactivated every power saving configs available on both laptops, tried 5ghz, everything set for performance, no solution.

While searching for a new mouse I came to understand Polling Rate, which is the update frequency of the mouse towards the pc.
If you have a good mouse, set for 500hz+ (in my case it was set for 1ms - 1000hz), this is the problem.

**As soon as I set my Polling Rate to 4ms (250hz) or 8ms (125hz), the problem was gone.** 
Anything above this, the Wi-Fi simply can't take it and the mouse starts stuttering on the virtual screen.

This was mentioned once on this topic and I remember trying it through regedit without really understanding why. 
Turns out I either 1. didn't reboot after changing for testing, or 2. Regedit couldn't override my mouse's hardware-saved config.

Hopefully this helps anyone going through this.
